hibernate-core/hibernate-core.gradle

```diff
@@ -203,6 +203,9 @@ xjc {
     }
 }
 
+generateGrammarSource {
+    arguments += "-traceParser"
```

Comment:
Is this going to affect the regular HQL parser as well?

Specifically I'm wondering about performance impact, if this is to help during development maybe it shoduld be commented out when done with debugging?

Reply:
The impact - if you look in the generated parsers, you will now see Antlr has added traceIn/traceOut calls at the start and end of each rule method.  I use that as hooks to debug "parse tree info".  This is extremely useful information for debugging parse problems; unless we can demonstrate that it is a performance problem I'd prefer to leave these as-is.

Reply:
Its a good point though in that the default impl Antlr injects is to use sysout.  For HQL and for SQL script parsing I've overridden these to use logging.  I'll add the same override for the entity-graph parser and really the impact overall should be minimal